config/routes.rb

```diff
@@ -1,6 +1,8 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  root to: "users#index"
+
+  resources :users, only: [:show] do
+    resources :posts, only: [:index, :show], param: :author_id
+  end
```

Comment:
- [OPTIONAL ] Apart from the root, Il would also be better to add the index to the users resources.
```ruby
  root to: "users#index"
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```